thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: "`7` is not a power of 2"'
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.